src/Uno.Toolkit.UI/Behaviors/CommandExtensions.cs

```diff
@@ -137,16 +149,54 @@ internal static bool TryInvokeCommand(DependencyObject owner, object? parameter)
 
 			return false;
 		}
+		internal static object? TryGetItemCommandParameter(DependencyObject? container)
+		{
+			if (container is ListViewItem)
+			{
+				// fixme: it doesn't work here, because we came from ListView::ItemClick,
+				// and when that happens the SelectedIndex not yet set.
+				return null;
+			}
+			if (container is { })
+			{
+				// we can have two scenarios here: // where the CommandParameter property can be assigned to
+				// 1. direct (IsItemItsOwnContainerOverride=true) container as items
+				if (GetCommandParameter(container) is { } parameter1)
+				{
+					return parameter1;
+				}
+
+				// 2. root element of item-template
+				if (container is ContentControl && // typically Selector's item-container are all of ContentControl descents: LVI, CBI, LBI...
+					container.GetFirstDescendant<ContentPresenter>(IsTemplateBoundToContent) is { } presenter &&
+					presenter.GetTemplateRoot() is { } root &&
+					GetCommandParameter(root) is { } parameter2)
+				{
+					return parameter2;
+				}
+
+				bool IsTemplateBoundToContent(ContentPresenter presenter) =>
+					presenter.GetBindingExpression(ContentPresenter.ContentProperty) is { ParentBinding.Path.Path: "Content" };
+			}
+
+			return null;
+		}
 
 		private static void OnListViewItemClick(object sender, ItemClickEventArgs e)
 		{
 			if (sender is not ListViewBase host) return;
 
-			TryInvokeCommand(host, GetCommandParameter(host) ?? e.ClickedItem);
+			TryInvokeCommand(host, /*TryGetItemCommandParameter(host.ContainerFromIndex(host.SelectedIndex)) ??*/ e.ClickedItem);
```

Comment:
TryGetItemCommandParameter commented here because?

Reply:
because of the fixme in TryGetItemCommandParameter